src/lib/Libcmds/pbs_json.c

```diff
@@ -160,7 +182,24 @@ strdup_escape(JsonEscapeType esc_type, const char *str)
 				break;
 			case '\\':
 				if (esc_type == JSON_ESCAPE) {
-					buf[i++] = *str++;
+					bufstr = str + 1;
+					if (*bufstr && ((*bufstr == '\'') || (*bufstr == ','))) {
+						int slash_ct = 0;
+						bufstr--;
+						while ((bufstr >= orig_str) && (*bufstr == '\\')) {
+							slash_ct++;
+							bufstr--;
+						}
+						/* detected balanced escaping (even # slashes) */
+						if ((slash_ct % 2) == 0) {
+							buf[i++] = *str++;
+						} else {
+							str++;
+						}
+
+					} else {
+						buf[i++] = *str++;
+					}
```

Comment:
can we handle the backslashes in pairs?
if you encounter a backslash, check the next character: if it's a comma/quote, only add the comma/quote.
Otherwise, add the backslash and next character.
This seems it would be simpler

Reply:
I can't do what you suggested since the function parses the input string character by character, and it will operate only the current character, and not the next character like if it's a comma/quote. So if it encounters a backslash, function is deciding to include the backslash, not to include (if followed by comma or quote where the preceding backslash is not escaped), or add an additional backslash. It's not operating on the next character...

Reply:
The function takes a string and returns it, right?
I see the while loop checks `(*str)`, we increment str once in each case. I'm saying we increment it twice in this one.

Reply:
@vstumpf Got it. I'll adjust the fix.